torch/csrc/jit/python/init.cpp

```diff
@@ -1044,6 +1044,26 @@ void initJITBindings(PyObject* module) {
         return self.getAllRecords();
       });
 
+  // Used by torch.Package to coordinate deserialization of storages across
+  // ScriptModules and eager modules
+  py::class_<StorageContextTracker, std::shared_ptr<StorageContextTracker>>(
+      m, "StorageContextTracker")
+      .def(py::init<>())
+      .def(
+          "get_tensor_impl",
+          [](StorageContextTracker& self, const std::string& name) {
+            auto tensor_impl = self.getStorage(name);
+            return at::Tensor(std::move(tensor_impl));
+          })
+      .def(
+          "add_tensor_impl",
+          [](StorageContextTracker& self,
+             const std::string& name,
+             const at::Tensor& tensor) {
```

Comment:
I'm not sure why, but I could only get `at::Tensor` and not `c10::Storage` to pass between Python and Cpp. The rest of the deserialization code looks as if it had the same issue. When looking at the uses of get_storage_from_record (the fucntion right above this one which is used for deserialization), people seem to use the tensor just as a vehicle for the storage also:
```
# In torch/serialization.py
 storage = zip_file.get_storage_from_record(name, size, dtype).storage()
```

Reply:
I think this is because c10::Storage is bound directly to Python as opposed to using pybind, so the magic "just return it and we'll figure out how to turn it into the right python type" doesn't work. That's fine, it's probably not worth untangling rn.